src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,131 @@
+import { useContext, useMemo } from 'react';
+import { PersonItem } from '../PersonItem';
+import { FilteredPeopleContext } from '../../peopleContext';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { SortBy, SortOrder, SortQueries } from '../../types/Sort';
+import { Person } from '../../types';
+import { sortPeople } from '../../utils/sortPeople';
+
+export const PeopleTable: React.FC = () => {
+  const [searchParams] = useSearchParams();
+  const filteredPeople = useContext(FilteredPeopleContext);
+  const navigate = useNavigate();
+
+  const sortByField = searchParams.get(SortQueries.sort);
+  const sortOrder = searchParams.get(SortQueries.order) || SortOrder.off;
+
+  const sortedPeople = useMemo(
+    () => sortPeople(filteredPeople, sortByField as keyof Person, sortOrder),
+    [filteredPeople, sortByField, sortOrder],
+  );
+
+  const sortHandler = (sortField: SortBy) => {
+    let tempSortOrder = SortOrder.off;
+
+    if (sortField === sortByField) {
+      switch (sortOrder) {
+        case SortOrder.off:
+          tempSortOrder = SortOrder.asc;
+          break;
+        case SortOrder.asc:
+          tempSortOrder = SortOrder.decs;
+          break;
+      }
```

Comment:
In the switch-case logic, you have not provided a default case. It's a good practice to always provide a default case in a switch statement to handle unexpected values. Here, if the `sortOrder` is neither `SortOrder.off` nor `SortOrder.asc`, there will be no matched case. Although it's unlikely to happen in your current code, it's better to be safe and handle such scenarios. Also, there seems to be a typo in line 31, it should be `SortOrder.desc` instead of `SortOrder.decs`.